diff -ur globus-core-5.15.orig/configure.in globus-core-5.15/configure.in
--- globus-core-5.15.orig/configure.in	2008-12-16 05:40:04.373821611 +0000
+++ globus-core-5.15/configure.in	2008-12-16 05:38:25.741802000 +0000
@@ -179,7 +179,6 @@
 
 
 AM_PROG_LIBTOOL
-mv ./libtool libtool-$GLOBUS_FLAVOR_NAME
 
 dnl
 dnl Output selected thread package
@@ -222,3 +221,4 @@
 	pkgdata/build-parameters
 	)
 
+mv ./libtool libtool-$GLOBUS_FLAVOR_NAME
