diff -ur globus_gridftp_server-3.15.orig/pkgdata/pkg_data_src.gpt.in globus_gridftp_server-3.15/pkgdata/pkg_data_src.gpt.in
--- globus_gridftp_server-3.15.orig/pkgdata/pkg_data_src.gpt.in	2008-09-27 01:05:50.000000000 +0200
+++ globus_gridftp_server-3.15/pkgdata/pkg_data_src.gpt.in	2009-07-28 08:56:52.238706022 +0200
@@ -22,29 +22,21 @@
                     <Simple_Version Major="0"/>
                 </Version>
             </Dependency>
-        </Source_Dependencies>
-        <Source_Dependencies Type="compile">
             <Dependency Name="globus_gridftp_server_control">
                 <Version>
                     <Simple_Version Major="0"/>
                 </Version>
             </Dependency>
-        </Source_Dependencies>
-        <Source_Dependencies Type="compile">
             <Dependency Name="globus_ftp_control">
                 <Version>
                     <Simple_Version Major="1"/>
                 </Version>
             </Dependency>
-        </Source_Dependencies>
-        <Source_Dependencies Type="compile">
             <Dependency Name="globus_authz">
                 <Version>
                     <Simple_Version Major="0"/>
                 </Version>
             </Dependency>
-        </Source_Dependencies>
-        <Source_Dependencies Type="compile">
             <Dependency Name="globus_usage">
                 <Version>
                     <Simple_Version Major="1"/>
@@ -68,30 +60,59 @@
                     <Simple_Version Major="0"/>
                 </Version>
             </Dependency>
-
-        </Source_Dependencies>
-        <Source_Dependencies Type="pgm_link">
             <Dependency Name="globus_gridftp_server_control"> 
                 <Version>
                     <Simple_Version Major="0"/>
                 </Version>
             </Dependency>
-        </Source_Dependencies>
-        <Source_Dependencies Type="pgm_link">
             <Dependency Name="globus_ftp_control"> 
                 <Version>
                     <Simple_Version Major="1"/>
                 </Version>
             </Dependency>
-        </Source_Dependencies>
-        <Source_Dependencies Type="pgm_link">
             <Dependency Name="globus_authz"> 
                 <Version>
                     <Simple_Version Major="0"/>
                 </Version>
             </Dependency>
+            <Dependency Name="globus_usage"> 
+                <Version>
+                    <Simple_Version Major="1"/>
+                </Version>
+            </Dependency>
         </Source_Dependencies>
-        <Source_Dependencies Type="pgm_link">
+
+        <Source_Dependencies Type="lib_link">
+            <Dependency Name="globus_xio"> 
+                <Version>
+                    <Simple_Version Major="0"/>
+                </Version>
+            </Dependency>
+            <Dependency Name="globus_xio_gsi_driver">
+                <Version>
+                    <Simple_Version Major="0"/>
+                </Version>
+            </Dependency>
+           <Dependency Name="globus_gfork">
+                <Version>
+                    <Simple_Version Major="0"/>
+                </Version>
+            </Dependency>
+            <Dependency Name="globus_gridftp_server_control"> 
+                <Version>
+                    <Simple_Version Major="0"/>
+                </Version>
+            </Dependency>
+            <Dependency Name="globus_ftp_control"> 
+                <Version>
+                    <Simple_Version Major="1"/>
+                </Version>
+            </Dependency>
+            <Dependency Name="globus_authz"> 
+                <Version>
+                    <Simple_Version Major="0"/>
+                </Version>
+            </Dependency>
             <Dependency Name="globus_usage"> 
                 <Version>
                     <Simple_Version Major="1"/>
@@ -105,6 +126,7 @@
             <cflags>@GPT_PKG_CFLAGS@</cflags>
             <external_includes>@GPT_EXTERNAL_INCLUDES@</external_includes>
             <external_libs>@GPT_EXTERNAL_LIBS@</external_libs>
+	    <pkg_libs>-lglobus_server</pkg_libs>
         </Build_Environment>
     </src_pkg>
 </gpt_package_metadata>
