--- xacml_auth_service.gsoap.old	2007-08-27 15:48:05.000000000 -0400
+++ xacml_auth_service.gsoap	2007-08-27 15:43:15.000000000 -0400
@@ -2249,7 +2249,7 @@
 };
 
 /// "urn:oasis:names:tc:SAML:2.0:assertion":AuthnStatementType is a complexType with complexContent extension of "urn:oasis:names:tc:SAML:2.0:assertion":StatementAbstractType.
-class saml__AuthnStatementType
+class saml__AuthnStatementType : public saml__StatementAbstractType
 { public:
 /// INHERITED FROM saml__StatementAbstractType:
 //  END OF INHERITED
@@ -2266,7 +2266,7 @@
 };
 
 /// "urn:oasis:names:tc:SAML:2.0:assertion":AuthzDecisionStatementType is a complexType with complexContent extension of "urn:oasis:names:tc:SAML:2.0:assertion":StatementAbstractType.
-class saml__AuthzDecisionStatementType
+class saml__AuthzDecisionStatementType : public saml__StatementAbstractType
 { public:
 /// INHERITED FROM saml__StatementAbstractType:
 //  END OF INHERITED
@@ -2290,7 +2290,7 @@
 };
 
 /// "urn:oasis:names:tc:SAML:2.0:assertion":AttributeStatementType is a complexType with complexContent extension of "urn:oasis:names:tc:SAML:2.0:assertion":StatementAbstractType.
-class saml__AttributeStatementType
+class saml__AttributeStatementType : public saml__StatementAbstractType
 { public:
 /// INHERITED FROM saml__StatementAbstractType:
 //  END OF INHERITED
@@ -2383,7 +2383,7 @@
 };
 
 /// "urn:oasis:xacml:2.0:saml:assertion:schema:os":XACMLAuthzDecisionStatementType is a complexType with complexContent extension of "urn:oasis:names:tc:SAML:2.0:assertion":StatementAbstractType.
-class XACMLassertion__XACMLAuthzDecisionStatementType
+class XACMLassertion__XACMLAuthzDecisionStatementType : public saml__StatementAbstractType
 { public:
 /// INHERITED FROM saml__StatementAbstractType:
 //  END OF INHERITED
@@ -2394,7 +2394,7 @@
 };
 
 /// "urn:oasis:xacml:2.0:saml:assertion:schema:os":XACMLPolicyStatementType is a complexType with complexContent extension of "urn:oasis:names:tc:SAML:2.0:assertion":StatementAbstractType.
-class XACMLassertion__XACMLPolicyStatementType
+class XACMLassertion__XACMLPolicyStatementType : public saml__StatementAbstractType
 { public:
 /// INHERITED FROM saml__StatementAbstractType:
 //  END OF INHERITED
