Index: source-trees/xio/src/builtins/gsi/Makefile.am
===================================================================
RCS file: /home/globdev/CVS/globus-packages/xio/src/builtins/gsi/Makefile.am,v
retrieving revision 1.2
diff -u -r1.2 Makefile.am
--- source-trees/xio/src/builtins/gsi/Makefile.am	21 Apr 2006 19:10:34 -0000	1.2
+++ source-trees/xio/src/builtins/gsi/Makefile.am	9 Feb 2010 02:07:26 -0000
@@ -1,25 +1,17 @@
 include $(top_srcdir)/globus_automake_pre
-include $(top_srcdir)/globus_automake_pre_top
-
-SUBDIRS = \
-	pkgdata \
-	doxygen
 
 flavorinclude_HEADERS = \
     globus_xio_gsi.h
 
 INCLUDES = $(GPT_INCLUDES)
 
-lib_LTLIBRARIES = libglobus_xio_gsi_driver_$(GLOBUS_FLAVOR_NAME).la
+noinst_LTLIBRARIES = libglobus_xio_gsi_driver_$(GLOBUS_FLAVOR_NAME).la
 
 libglobus_xio_gsi_driver___GLOBUS_FLAVOR_NAME__la_LIBADD = \
 	@GPT_LIB_LINKS@
 libglobus_xio_gsi_driver___GLOBUS_FLAVOR_NAME__la_SOURCES = \
 	globus_i_xio_gsi.h  globus_xio_gsi.h globus_xio_gsi.c
 libglobus_xio_gsi_driver___GLOBUS_FLAVOR_NAME__la_LDFLAGS = \
-	@GPT_LDFLAGS@ -module -rpath $(GLOBUS_LOCATION)/lib
-
-EXTRA_DIST = version.h.in
+	@GPT_LDFLAGS@ 
 
 include $(top_srcdir)/globus_automake_post
-include $(top_srcdir)/globus_automake_post_top
Index: source-trees/xio/src/builtins/pipe/Makefile.am
===================================================================
RCS file: /home/globdev/CVS/globus-packages/xio/src/builtins/pipe/Makefile.am,v
retrieving revision 1.1
diff -u -r1.1 Makefile.am
--- source-trees/xio/src/builtins/pipe/Makefile.am	15 May 2009 01:13:38 -0000	1.1
+++ source-trees/xio/src/builtins/pipe/Makefile.am	9 Feb 2010 02:07:26 -0000
@@ -1,24 +1,19 @@
 include $(top_srcdir)/globus_automake_pre
-include $(top_srcdir)/globus_automake_pre_top
 
-SUBDIRS = \
-	pkgdata 
 
 flavorinclude_HEADERS = \
     globus_xio_pipe_driver.h
 
 INCLUDES = $(GPT_INCLUDES)
 
-lib_LTLIBRARIES = libglobus_xio_pipe_driver_$(GLOBUS_FLAVOR_NAME).la
+noinst_LTLIBRARIES = libglobus_xio_pipe_driver_$(GLOBUS_FLAVOR_NAME).la
 
 libglobus_xio_pipe_driver___GLOBUS_FLAVOR_NAME__la_LIBADD = \
 	@GPT_LIB_LINKS@
 libglobus_xio_pipe_driver___GLOBUS_FLAVOR_NAME__la_SOURCES = \
 	globus_xio_pipe_driver.c
 libglobus_xio_pipe_driver___GLOBUS_FLAVOR_NAME__la_LDFLAGS = \
-	@GPT_LDFLAGS@ -module -rpath $(GLOBUS_LOCATION)/lib
+	@GPT_LDFLAGS@ 
 
-EXTRA_DIST = version.h.in
 
 include $(top_srcdir)/globus_automake_post
-include $(top_srcdir)/globus_automake_post_top
Index: source-trees/xio/src/builtins/popen/Makefile.am
===================================================================
RCS file: /home/globdev/CVS/globus-packages/xio/src/builtins/popen/Makefile.am,v
retrieving revision 1.1
diff -u -r1.1 Makefile.am
--- source-trees/xio/src/builtins/popen/Makefile.am	8 Sep 2006 17:15:29 -0000	1.1
+++ source-trees/xio/src/builtins/popen/Makefile.am	9 Feb 2010 02:07:26 -0000
@@ -1,24 +1,19 @@
 include $(top_srcdir)/globus_automake_pre
-include $(top_srcdir)/globus_automake_pre_top
 
-SUBDIRS = \
-	pkgdata 
 
 flavorinclude_HEADERS = \
     globus_xio_popen_driver.h
 
 INCLUDES = $(GPT_INCLUDES)
 
-lib_LTLIBRARIES = libglobus_xio_popen_driver_$(GLOBUS_FLAVOR_NAME).la
+noinst_LTLIBRARIES = libglobus_xio_popen_driver_$(GLOBUS_FLAVOR_NAME).la
 
 libglobus_xio_popen_driver___GLOBUS_FLAVOR_NAME__la_LIBADD = \
 	@GPT_LIB_LINKS@
 libglobus_xio_popen_driver___GLOBUS_FLAVOR_NAME__la_SOURCES = \
 	globus_xio_popen_driver.c
 libglobus_xio_popen_driver___GLOBUS_FLAVOR_NAME__la_LDFLAGS = \
-	@GPT_LDFLAGS@ -module -rpath $(GLOBUS_LOCATION)/lib
+	@GPT_LDFLAGS@ 
 
-EXTRA_DIST = version.h.in
 
 include $(top_srcdir)/globus_automake_post
-include $(top_srcdir)/globus_automake_post_top
Index: source-trees/xio/src/pkgdata/pkg_data_src.gpt.in
===================================================================
RCS file: /home/globdev/CVS/globus-packages/xio/src/pkgdata/pkg_data_src.gpt.in,v
retrieving revision 1.53
diff -u -r1.53 pkg_data_src.gpt.in
--- source-trees/xio/src/pkgdata/pkg_data_src.gpt.in	16 Jul 2009 19:27:47 -0000	1.53
+++ source-trees/xio/src/pkgdata/pkg_data_src.gpt.in	9 Feb 2010 02:07:26 -0000
@@ -15,10 +15,28 @@
     <Source_Dependencies Type="compile" >
       <Dependency Name="globus_common" > <Version > <Simple_Version Major="4" /> </Version> </Dependency>
       <Dependency Name="globus_core" > <Version > <Simple_Version Major="4" /> </Version> </Dependency>
+            <Dependency Name="globus_gss_assist" >
+                <Version><Simple_Version Major="3"/></Version> 
+            </Dependency>
+            <Dependency Name="gssapi_error">
+                <Version ><Simple_Version Major="2"/></Version>
+            </Dependency>
+            <Dependency Name="globus_gssapi_gsi"> 
+                <Version ><Simple_Version Major="4"/></Version>
+            </Dependency>
     </Source_Dependencies>
 
     <Source_Dependencies Type="lib_link" >
       <Dependency Name="globus_common" > <Version > <Simple_Version Major="4" /> </Version> </Dependency>
+            <Dependency Name="globus_gss_assist" >
+                <Version><Simple_Version Major="3"/></Version> 
+            </Dependency>
+            <Dependency Name="gssapi_error">
+                <Version ><Simple_Version Major="2"/></Version>
+            </Dependency>
+            <Dependency Name="globus_gssapi_gsi"> 
+                <Version ><Simple_Version Major="4"/></Version>
+            </Dependency>
     </Source_Dependencies>
 
     <Build_Environment >
