Index: source-trees/xio/src/builtins/gsi/Makefile.am
===================================================================
RCS file: /home/globdev/CVS/globus-packages/xio/src/builtins/gsi/Makefile.am,v
retrieving revision 1.2
diff -u -r1.2 Makefile.am
--- source-trees/xio/src/builtins/gsi/Makefile.am	21 Apr 2006 19:10:34 -0000	1.2
+++ source-trees/xio/src/builtins/gsi/Makefile.am	9 Feb 2010 02:07:26 -0000
@@ -1,25 +1,17 @@
 include $(top_srcdir)/globus_automake_pre
-include $(top_srcdir)/globus_automake_pre_top
-
-SUBDIRS = \
-	pkgdata \
-	doxygen
 
 flavorinclude_HEADERS = \
     globus_xio_gsi.h
 
 INCLUDES = $(GPT_INCLUDES)
 
-lib_LTLIBRARIES = libglobus_xio_gsi_driver_$(GLOBUS_FLAVOR_NAME).la
+noinst_LTLIBRARIES = libglobus_xio_gsi_driver_$(GLOBUS_FLAVOR_NAME).la
 
 libglobus_xio_gsi_driver___GLOBUS_FLAVOR_NAME__la_LIBADD = \
 	@GPT_LIB_LINKS@
 libglobus_xio_gsi_driver___GLOBUS_FLAVOR_NAME__la_SOURCES = \
 	globus_i_xio_gsi.h  globus_xio_gsi.h globus_xio_gsi.c
 libglobus_xio_gsi_driver___GLOBUS_FLAVOR_NAME__la_LDFLAGS = \
-	@GPT_LDFLAGS@ -module -rpath $(GLOBUS_LOCATION)/lib
-
-EXTRA_DIST = version.h.in
+	@GPT_LDFLAGS@ 
 
 include $(top_srcdir)/globus_automake_post
-include $(top_srcdir)/globus_automake_post_top
Index: source-trees/xio/src/builtins/pipe/Makefile.am
===================================================================
RCS file: /home/globdev/CVS/globus-packages/xio/src/builtins/pipe/Makefile.am,v
retrieving revision 1.1
diff -u -r1.1 Makefile.am
--- source-trees/xio/src/builtins/pipe/Makefile.am	15 May 2009 01:13:38 -0000	1.1
+++ source-trees/xio/src/builtins/pipe/Makefile.am	9 Feb 2010 02:07:26 -0000
@@ -1,24 +1,19 @@
 include $(top_srcdir)/globus_automake_pre
-include $(top_srcdir)/globus_automake_pre_top
 
-SUBDIRS = \
-	pkgdata 
 
 flavorinclude_HEADERS = \
     globus_xio_pipe_driver.h
 
 INCLUDES = $(GPT_INCLUDES)
 
-lib_LTLIBRARIES = libglobus_xio_pipe_driver_$(GLOBUS_FLAVOR_NAME).la
+noinst_LTLIBRARIES = libglobus_xio_pipe_driver_$(GLOBUS_FLAVOR_NAME).la
 
 libglobus_xio_pipe_driver___GLOBUS_FLAVOR_NAME__la_LIBADD = \
 	@GPT_LIB_LINKS@
 libglobus_xio_pipe_driver___GLOBUS_FLAVOR_NAME__la_SOURCES = \
 	globus_xio_pipe_driver.c
 libglobus_xio_pipe_driver___GLOBUS_FLAVOR_NAME__la_LDFLAGS = \
-	@GPT_LDFLAGS@ -module -rpath $(GLOBUS_LOCATION)/lib
+	@GPT_LDFLAGS@ 
 
-EXTRA_DIST = version.h.in
 
 include $(top_srcdir)/globus_automake_post
-include $(top_srcdir)/globus_automake_post_top
Index: source-trees/xio/src/builtins/popen/Makefile.am
===================================================================
RCS file: /home/globdev/CVS/globus-packages/xio/src/builtins/popen/Makefile.am,v
retrieving revision 1.1
diff -u -r1.1 Makefile.am
--- source-trees/xio/src/builtins/popen/Makefile.am	8 Sep 2006 17:15:29 -0000	1.1
+++ source-trees/xio/src/builtins/popen/Makefile.am	9 Feb 2010 02:07:26 -0000
@@ -1,24 +1,19 @@
 include $(top_srcdir)/globus_automake_pre
-include $(top_srcdir)/globus_automake_pre_top
 
-SUBDIRS = \
-	pkgdata 
 
 flavorinclude_HEADERS = \
     globus_xio_popen_driver.h
 
 INCLUDES = $(GPT_INCLUDES)
 
-lib_LTLIBRARIES = libglobus_xio_popen_driver_$(GLOBUS_FLAVOR_NAME).la
+noinst_LTLIBRARIES = libglobus_xio_popen_driver_$(GLOBUS_FLAVOR_NAME).la
 
 libglobus_xio_popen_driver___GLOBUS_FLAVOR_NAME__la_LIBADD = \
 	@GPT_LIB_LINKS@
 libglobus_xio_popen_driver___GLOBUS_FLAVOR_NAME__la_SOURCES = \
 	globus_xio_popen_driver.c
 libglobus_xio_popen_driver___GLOBUS_FLAVOR_NAME__la_LDFLAGS = \
-	@GPT_LDFLAGS@ -module -rpath $(GLOBUS_LOCATION)/lib
+	@GPT_LDFLAGS@ 
 
-EXTRA_DIST = version.h.in
 
 include $(top_srcdir)/globus_automake_post
-include $(top_srcdir)/globus_automake_post_top
Index: source-trees/xio/src/pkgdata/pkg_data_src.gpt.in
===================================================================
RCS file: /home/globdev/CVS/globus-packages/xio/src/pkgdata/pkg_data_src.gpt.in,v
retrieving revision 1.53
diff -u -r1.53 pkg_data_src.gpt.in
--- source-trees/xio/src/pkgdata/pkg_data_src.gpt.in	16 Jul 2009 19:27:47 -0000	1.53
+++ source-trees/xio/src/pkgdata/pkg_data_src.gpt.in	9 Feb 2010 02:07:26 -0000
@@ -15,10 +15,28 @@
     <Source_Dependencies Type="compile" >
       <Dependency Name="globus_common" > <Version > <Simple_Version Major="4" /> </Version> </Dependency>
       <Dependency Name="globus_core" > <Version > <Simple_Version Major="4" /> </Version> </Dependency>
+            <Dependency Name="globus_gss_assist" >
+                <Version><Simple_Version Major="3"/></Version> 
+            </Dependency>
+            <Dependency Name="gssapi_error">
+                <Version ><Simple_Version Major="2"/></Version>
+            </Dependency>
+            <Dependency Name="globus_gssapi_gsi"> 
+                <Version ><Simple_Version Major="4"/></Version>
+            </Dependency>
     </Source_Dependencies>
 
     <Source_Dependencies Type="lib_link" >
       <Dependency Name="globus_common" > <Version > <Simple_Version Major="4" /> </Version> </Dependency>
+            <Dependency Name="globus_gss_assist" >
+                <Version><Simple_Version Major="3"/></Version> 
+            </Dependency>
+            <Dependency Name="gssapi_error">
+                <Version ><Simple_Version Major="2"/></Version>
+            </Dependency>
+            <Dependency Name="globus_gssapi_gsi"> 
+                <Version ><Simple_Version Major="4"/></Version>
+            </Dependency>
     </Source_Dependencies>
 
     <Build_Environment >
Index: source-trees/database/c/iodbc/patches/libiodbc-3.52.6-patch
===================================================================
RCS file: /home/globdev/CVS/globus-packages/database/c/iodbc/patches/libiodbc-3.52.6-patch,v
retrieving revision 1.2.2.1
diff -u -r1.2.2.1 libiodbc-3.52.6-patch
--- source-trees/database/c/iodbc/patches/libiodbc-3.52.6-patch 25 Jan 2011 22:17:17 -0000      1.2.2.1
+++ source-trees/database/c/iodbc/patches/libiodbc-3.52.6-patch 3 Mar 2011 11:00:22 -0000
@@ -88,3 +88,24 @@
      bin/Makefile
      bin/iodbc-config
      etc/Makefile
+diff -u -r libiodbc-3.52.6/iodbc/info.c libiodbc-3.52.6-patched/iodbc/info.c
+--- libiodbc-3.52.6/iodbc/info.c  2007-01-05 06:22:50.000000000 -0600
++++ libiodbc-3.52.6-patched/iodbc/info.c  2011-03-02 00:14:11.379412200 -0600
+@@ -109,7 +109,7 @@
+ 
+ 
+ static int
+-stricmp (const char *s1, const char *s2)
++infostricmp (const char *s1, const char *s2)
+ {
+   int cmp;
+ 
+@@ -130,7 +130,7 @@
+   const char **s1 = (const char **) p1;
+   const char **s2 = (const char **) p2;
+ 
+-  return stricmp (*s1, *s2);
++  return infostricmp (*s1, *s2);
+ }
+ 
+ 
Index: source-trees/database/c/sqlite/patches/sqlite-3.3.17-patch
===================================================================
RCS file: source-trees/database/c/sqlite/patches/sqlite-3.3.17-patch
diff -N source-trees/database/c/sqlite/patches/sqlite-3.3.17-patch
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ source-trees/database/c/sqlite/patches/sqlite-3.3.17-patch	3 Mar 2011 12:33:58 -0000
@@ -0,0 +1,11 @@
+--- sqlite-3.3.17/Makefile.in  2007-04-06 07:44:37.000000000 -0500
++++ sqlite-3.3.17-patched/Makefile.in  2011-03-03 05:38:12.000000000 -0600
+@@ -664,7 +664,7 @@
+ 	mkdir -p doc
+ 	mv $(DOC) doc
+ 
+-install:	sqlite3 libsqlite3.la sqlite3.h ${HAVE_TCL:1=tcl_install}
++install:	sqlite3$(TEXE) libsqlite3.la sqlite3.h ${HAVE_TCL:1=tcl_install}
+ 	$(INSTALL) -d $(DESTDIR)$(libdir)
+ 	$(LTINSTALL) libsqlite3.la $(DESTDIR)$(libdir)
+ 	$(INSTALL) -d $(DESTDIR)$(exec_prefix)/bin
Index: source-trees/gsi/sysconfig/source/library/globus_gsi_system_config.c
===================================================================
RCS file: /home/globdev/CVS/globus-packages/gsi/sysconfig/source/library/globus_gsi_system_config.c,v
retrieving revision 1.83.4.2
diff -u -r1.83.4.2 globus_gsi_system_config.c
--- source-trees/gsi/sysconfig/source/library/globus_gsi_system_config.c     15 Feb 2010 01:21:00 -0000      1.83.4.2
+++ source-trees/gsi/sysconfig/source/library/globus_gsi_system_config.c     3 Apr 2011 23:23:22 -0000
@@ -230,6 +230,11 @@
 
 int globus_i_gsi_sysconfig_debug_level = 0;
 
+
+#ifdef TARGET_ARCH_CYGWIN
+static int globus_l_gsi_sysconfig_skip_check = 0;
+#endif
+
 /**
  * Module descriptor static initializer.
  */
@@ -291,6 +296,15 @@
 
     GLOBUS_I_GSI_SYSCONFIG_DEBUG_ENTER;
 
+#ifdef TARGET_ARCH_CYGWIN
+    /* windows permissions are a pain, and cygwin won't give reliable 
+    permissions when a non-posix path is used anyways */
+    if(getenv("GLOBUS_IGNORE_CRED_PERMS"))
+    {
+        globus_l_gsi_sysconfig_skip_check = 1;
+    }
+#endif
+
     result = globus_module_activate(GLOBUS_COMMON_MODULE);
 
     if(result != 0)
@@ -4786,6 +4800,13 @@
 
     GLOBUS_I_GSI_SYSCONFIG_DEBUG_ENTER;
 
+#ifdef TARGET_ARCH_CYGWIN
+    if(globus_l_gsi_sysconfig_skip_check)
+    {
+        return globus_gsi_sysconfig_file_exists_unix(filename);
+    }
+#endif
+
     if (stat(filename,&stx) == -1)
     {
         switch (errno)
@@ -4919,6 +4940,13 @@
         "globus_i_gsi_sysconfig_check_certfile_unix";
 
     GLOBUS_I_GSI_SYSCONFIG_DEBUG_ENTER;
+
+#ifdef TARGET_ARCH_CYGWIN
+    if(globus_l_gsi_sysconfig_skip_check)
+    {
+        return globus_gsi_sysconfig_file_exists_unix(filename);
+    }
+#endif
     
     if (stat(filename,&stx) == -1)
     {
